### PR TITLE
Ignore process selector warnings by default

### PR DESCRIPTION
**Pull request by fa2k**
## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] Documentation in `docs` is updated

This addresses issue #2161.

Changes nextflow.config in the template to set `nextflow.enable.configProcessNamesValidation = false`. The `debug` profile can then be used to re-enable these warnings.

The pull request template is updated to include a "testing with `debug` profile" step.

I will also send a PR for the website (docs).

### CODE REVIEW

File: nf_core/pipeline-template/nextflow.config [hunk 1]
```diff
@@ -216,6 +217,9 @@ env {
 // Capture exit codes from upstream processes when piping
 process.shell = ['/bin/bash', '-euo', 'pipefail']
 
+// Disable process selector warnings by default. Use debug profile to enable warnings.
+nextflow.enable.configProcessNamesValidation = false
```
Comment (mirpedrol) [inline nf_core/pipeline-template/nextflow.config:221]: No need to add this param in the global scope

**Review by mirpedrol — COMMENTED** [review]
Could you also modify the documentation in [CONTRIBUTING.md](https://github.com/nextflow-io/nextflow/blob/master/CONTRIBUTING.md)? Otherwise, looks good! :) 
And after this PR you can also add the documentation to the website, contributing documentation is [here](https://nf-co.re/docs/contributing/contributing_to_pipelines)

File: nf_core/pipeline-template/nextflow.config (hunk 1 above)
Comment (fa2k) [reply nf_core/pipeline-template/nextflow.config:221]: Thanks for the review!
* I've added these notes to the website - does it look okay? https://github.com/nf-core/website/pull/2040/files
* I will work on CONTRIBUTING.md

Comment (fa2k) [reply nf_core/pipeline-template/nextflow.config:221]: > No need to add this param in the global scope

The default for this setting in nextflow is true. My idea for this is to set it to false by default, then return it to true if using debug.

Comment (mirpedrol) [reply nf_core/pipeline-template/nextflow.config:221]: yes, sorry I missed that it was `true` by default. Your code looks good then

**Comment by fa2k** [comment]
I've added the same text in CONTRIBUTING.md, so it should be ready to go when the test is done :)

**Review by mirpedrol — APPROVED** [review]
LGTM when tests pass 🚀

**Comment by fa2k** [comment]
> LGTM when tests pass 🚀

Thanks! It seems to be failing due to rate limiting on some of the APIs used by the tests. pytest has passed on one platform, Python 3.11, but not the others. Is it okay to merge then?

**Comment by mirpedrol** [comment]
I've re-run the failed tests, let's see if they pass now

**Pull request merged**